Use wp_admin_notice

joedolson opened the pull request:
PR for finishing applying `wp_admin_notice` functions across the admin. In progress.

Trac ticket: https://core.trac.wordpress.org/ticket/57791

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

joedolson committed:
Authorize application & edit form blocks JS fallback

joedolson committed:
Update authorize-application.php

costdev commented on src/wp-admin/authorize-application.php:
```diff
@@ -194,24 +201,24 @@
+			<p>' . __( 'Be sure to save this in a safe location. You will not be able to retrieve it.' ) . '</p>';
+			$args = array(
+				'type'               => 'success',
+				'additional_classes' => array( 'notice-alt', 'below-h2' ),
```
I think we might need `'paragraph_wrap' => false` here as there's already `<p>` tags included in the message.

joedolson committed:
I'm about half way through; taking a break so I'm committing the work in progress.

joedolson committed:
Set p wrap false in authorize application

joedolson committed:
Continuing work

joedolson committed:
Update function `wp_update_php_annotation()` to allow it to return instead of echoing.

Change needed to support theme update errors, in same commit.

joedolson committed:
Think this finishes off the obvious ones.

costdev commented on src/wp-includes/functions.php:
```diff
@@ -8274,15 +8274,23 @@ function wp_get_default_update_php_url() {
+ * @return string|void
  */
-function wp_update_php_annotation( $before = '<p class="description">', $after = '</p>' ) {
+function wp_update_php_annotation( $before = '<p class="description">', $after = '</p>', $echo = true ) {
```
FYI in case you hadn't seen it: [Here's what I did](https://github.com/WordPress/wordpress-develop/pull/5149/files#diff-03c8ac6c0d90d91bd93d3aa0982efa4248f993cf4f009c225db938f2d64f505e) for these type of notices.

Happy for this function to be updated though if that makes more sense.

joedolson replied:
If there's an existing getter, that probably makes more sense.

joedolson replied:
On looking, the getter isn't equivalent, so I think this solution may be better.

joedolson committed:
misc. code standards changes

joedolson committed:
Additional CS changes

joedolson committed:
I'm now generating errors in places I didn't touch.

joedolson committed:
Merge branch 'WordPress:trunk' into apply-57791

joedolson committed:
Merge branch 'WordPress:trunk' into apply-57791

joedolson committed:
Merge branch 'WordPress:trunk' into apply-57791

joedolson committed:
Update plugin-editor.php

Mixing ternaries and concatenation is difficult to read and prone to operation order issues; assign the error message separately.

joedolson committed:
Alignment fix

joedolson committed:
Update theme-editor.php

Same ternary/operator issue as in plugin-editor

joedolson committed:
Stray opening div still in place

joedolson commented on src/wp-admin/theme-editor.php:
```diff
@@ -304,23 +317,30 @@
-				<?php endif; ?>
+				<?php
+				if ( is_child_theme() && $theme->get_stylesheet() === get_template() ) :
+					$message  = ( is_writable( $file ) ) : '<strong>' . __( 'Caution:' ) . '</strong> ' ? '';
```
Ternary operators reversed; fixed in commit.

joedolson commented:
Committed in r56570.
